salt/modules/win_lgpo.py

```diff
@@ -3555,72 +3572,80 @@ def _buildElementNsmap(using_elements):
-        return False, None
+    f_exp = os.path.join(__opts__['cachedir'], 'secedit-{0}.txt'.format(UUID))
+    __salt__['cmd.run'](['secedit', '/export', '/cfg', f_exp])
+    with io.open(f_exp, encoding='utf-16') as fp:
+        secedit_data = fp.readlines()
+    if __salt__['file.file_exists'](f_exp):
+        __salt__['file.remove'](f_exp)
+    return secedit_data
 
 
-def _importSeceditConfig(infdata):
+def _get_secedit_data():
     '''
-    helper function to write data to a temp file/run secedit to import policy/cleanup
+    Helper function that returns the secedit data in __context__ if it exists
+    and puts the secedit data in __context__ if it does not.
+
+    Returns:
+        str: secedit data from __context__
     '''
-    try:
-        _d = uuid.uuid4().hex
-        _tSdbfile = '{0}\\{1}'.format(__opts__['cachedir'],
-                                      'salt-secedit-import-{0}.sdb'.format(_d))
-        _tInfFile = '{0}\\{1}'.format(__opts__['cachedir'],
-                                      'salt-secedit-config-{0}.inf'.format(_d))
-        # make sure our temp files don't already exist
-        if __salt__['file.file_exists'](_tSdbfile):
-            _ret = __salt__['file.remove'](_tSdbfile)
-        if __salt__['file.file_exists'](_tInfFile):
-            _ret = __salt__['file.remove'](_tInfFile)
-        # add the inf data to the file, win_file sure could use the write() function
-        _ret = __salt__['file.touch'](_tInfFile)
-        _ret = __salt__['file.append'](_tInfFile, infdata)
-        # run secedit to make the change
-        _ret = __salt__['cmd.run']('secedit /configure /db {0} /cfg {1}'.format(_tSdbfile, _tInfFile))
-        # cleanup our temp files
-        if __salt__['file.file_exists'](_tSdbfile):
-            _ret = __salt__['file.remove'](_tSdbfile)
-        if __salt__['file.file_exists'](_tInfFile):
-            _ret = __salt__['file.remove'](_tInfFile)
+    if 'lgpo.secedit_data' not in __context__:
+        log.debug('LGPO: Loading secedit data')
+        __context__['lgpo.secedit_data'] = _load_secedit_data()
+    return __context__['lgpo.secedit_data']
+
+
+def _get_secedit_value(option):
+    '''
+    Helper function that looks for the passed option in the secedit data
+    '''
+    secedit_data = _get_secedit_data()
+    for _line in secedit_data:
+        if _line.startswith(option):
+            return _line.split('=')[1].strip()
+    return 'Not Defined'
+
+
+def _write_secedit_data(inf_data):
+    '''
+    Helper function to write secedit data to the database
+    '''
+    # Set file names
+    f_sdb = os.path.join(__opts__['cachedir'], 'secedit-{0}.sdb'.format(UUID))
+    f_inf = os.path.join(__opts__['cachedir'], 'secedit-{0}.inf'.format(UUID))
+
+    # Write the changes to the inf file
+    __salt__['file.write'](f_inf, inf_data)
+
+    # Run secedit to make the change
+    retcode = __salt__['cmd.retcode'](['secedit', '/configure', '/db', f_sdb,
```

Comment:
Should this be in a try/finally to make sure we do not leave stale files around?